exec/java-exec/src/test/java/org/apache/drill/exec/store/parquet/TestParquetMetadataCache.java

```diff
@@ -424,32 +432,121 @@ public void testMoveCache() throws Exception {
+      test("create table dfs_test.tmp.`%1$s/%1$s` as select * from cp.`tpch/nation.parquet`", pathWithSpaces);
+      test("refresh table metadata dfs_test.tmp.`%s`", pathWithSpaces);
+      checkForMetadataFile(pathWithSpaces);
+      String query = String.format("select * from dfs_test.tmp.`%s`", pathWithSpaces);
+      int expectedRowCount = 50;
+      int expectedNumFiles = 1; // point to selectionRoot since no pruning is done in this query
+      int actualRowCount = testSql(query);
+      assertEquals("An incorrect result was obtained while querying a table with metadata cache files",
+          expectedRowCount, actualRowCount);
+      String numFilesPattern = "numFiles=" + expectedNumFiles;
+      String usedMetaPattern = "usedMetadataFile=true";
+      String cacheFileRootPattern = String.format("cacheFileRoot=%s/%s", getDfsTestTmpSchemaLocation(), pathWithSpaces);
+      PlanTestBase.testPlanMatchingPatterns(query, new String[]{numFilesPattern, usedMetaPattern, cacheFileRootPattern},
+          new String[] {"Filter"});
+    } finally {
+      test("drop table if exists dfs_test.tmp.`%s`", pathWithSpaces);
+    }
+  }
+
+  @Test
+  public void testFutureUnsupportedMetadataVersion() throws Exception {
+    final String unsupportedMetadataVersion = "unsupported_metadata_version";
+    try {
+      test("use dfs_test.tmp");
+      test("create table `%s` as select * from cp.`tpch/nation.parquet`", unsupportedMetadataVersion);
+      MetadataVersion lastVersion = Iterables.getLast(MetadataVersion.Constants.SUPPORTED_VERSIONS);
+      for (MetadataVersion supportedVersion : MetadataVersion.Constants.SUPPORTED_VERSIONS) {
+        if (lastVersion.compareTo(supportedVersion) < 0) {
+          lastVersion = supportedVersion;
+        }
+      }
+      // Get the future version, which is absent in MetadataVersion.SUPPORTED_VERSIONS list
+      String futureVersion = "v" + (Integer.parseInt(String.valueOf(lastVersion.toString().charAt(1))) + 1);
+      copyMetaDataCacheToTempWithReplacements("parquet/unsupported_metadata/unsupported_metadata_version.requires_replace.txt",
+          unsupportedMetadataVersion, Metadata.METADATA_FILENAME, futureVersion);
+      String query = String.format("select * from %s", unsupportedMetadataVersion);
+      int expectedRowCount = 25;
+      int expectedNumFiles = 1;
+      int actualRowCount = testSql(query);
+      assertEquals("An incorrect result was obtained while querying a table with metadata cache files",
+          expectedRowCount, actualRowCount);
+      String numFilesPattern = "numFiles=" + expectedNumFiles;
+      String usedMetaPattern = "usedMetadataFile=false"; // ignoring metadata cache file
+      PlanTestBase.testPlanMatchingPatterns(query, new String[]{numFilesPattern, usedMetaPattern},
+          new String[] {"Filter"});
+    } finally {
+      test("drop table if exists %s", unsupportedMetadataVersion);
+    }
+  }
+
+  @Test
+  public void testCorruptedMetadataFile() throws Exception {
```

Comment:
Please add tests for the empty metadata file and when at lest one metadata file is missing.